src/views/CountryThreeWProjects/index.tsx

```diff
@@ -0,0 +1,515 @@
+import Container from '#components/Container';
+import Link from '#components/Link';
+import Button from '#components/Button';
+import PieChart from '#components/PieChart';
+import BlockLoading from '#components/BlockLoading';
+import KeyFigure from '#components/KeyFigure';
+import { useRequest } from '#utils/restRequest';
+import { sumSafe, denormalizeList } from '#utils/common';
+import type { CountryOutletContext } from '#utils/outletContext';
+import useTranslation from '#hooks/useTranslation';
+import ExpandableContainer from '#components/ExpandableContainer';
+import { resolveToString } from '#utils/translation';
+import {
+    numericValueSelector,
+    stringLabelSelector,
+    numericIdSelector,
+} from '#utils/selectors';
+import type { GoApiResponse } from '#utils/restRequest';
+import { type components } from '#generated/types';
+import RouteContext from '#contexts/route';
+
+import ProjectActions from './ProjectActions';
+import Map from './Map';
+import Filter, { FilterValue } from './Filters';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+interface LabelValue {
+    label: string;
+    value: number;
+}
+
+type District = NonNullable<GoApiResponse<'/api/v2/district/'>['results']>[number];
+type Project = NonNullable<GoApiResponse<'/api/v2/project/'>['results']>[number];
+
+type ProjectKey = keyof Project;
+
+// const PROJECT_STATUS_COMPLETED = 2;
+const PROJECT_STATUS_ONGOING = 1 satisfies components['schemas']['Status1d2Enum'];
+// const PROJECT_STATUS_PLANNED = 0;
+
+const emptyDistrictList: District[] = [];
+const emptyProjectList: Project[] = [];
+
+const primaryRedColorShades = [
+    'var(--go-ui-color-red-90)',
+    'var(--go-ui-color-red-60)',
+    'var(--go-ui-color-red-40)',
+    'var(--go-ui-color-red-20)',
+    'var(--go-ui-color-red-10)',
+];
+
+function filterProjects(
+    projectList: Project[],
+    filters: Partial<Record<ProjectKey, number[]>>,
+) {
+    const filterKeys = Object.keys(filters) as ProjectKey[];
+
+    return filterKeys.reduce((filteredProjectList, filterKey) => {
+        const filterValue = filters[filterKey];
+
+        if (filterValue?.length === 0) {
+            return filteredProjectList;
+        }
+
+        return filteredProjectList.filter((project) => {
+            const value = project[filterKey];
+
+            if (isNotDefined(value)) {
+                return true;
+            }
+
+            if (isNotDefined(filterValue)) {
+                return true;
+            }
+
+            if (Array.isArray(value)) {
+                return value.some(
+                    (v) => (
+                        filterValue?.findIndex(
+                            (fv) => String(fv) === String(v),
+                        ) !== -1
+                    ),
+                );
+            }
+
+            return filterValue?.findIndex(
+                (fv) => String(fv) === String(value),
+            ) !== -1;
+        });
+    }, projectList);
```

Comment:
IMO, it would be easier to first iterate over the project list
then define a filter function